dev/fattest.simplicity/src/componenttest/containers/ArtifactoryImageNameSubstitutor.java

```diff
@@ -56,33 +56,33 @@ public DockerImageName apply(final DockerImageName original) {
 
-            // Priority 2: If registry was explicit set, do not substitute
+            // Priority 2: If the image is known to only exist in an Artifactory registry
+            if (original.getRegistry() != null && original.getRegistry().contains("artifactory.swg-devops.com")) {
+                throw new RuntimeException("Not all developers of Open Liberty have access to artifactory, must use a public registry.");
+            }
+
+            // Priority 3: If a public registry was explicitly set on an image, do not substitute
             if (original.getRegistry() != null && !original.getRegistry().isEmpty()) {
                 result = original;
                 reason = "Image name is explicitally set with registry, cannot modify registry.";
                 break;
             }
 
-            // Priority 3: If the image is known to only exist in an Artifactory organization
-            // No need to check for this in Open Liberty since all images need to be accessible outside of Artifactory.
-
-            // Priority 4: System property use.artifactory.substitution (NOTE: only honor this property if set to true)
-            if (Boolean.getBoolean(substitutionOverride)) {
+            // Priority 4: Always use Artifactory if using remote docker host.
+            if (DockerClientFactory.instance().isUsing(EnvironmentAndSystemPropertyClientProviderStrategy.class)) {
                 result = DockerImageName.parse(mirror + '/' + original.asCanonicalNameString())
                                 .withRegistry(ArtifactoryRegistry.instance().getRegistry())
                                 .asCompatibleSubstituteFor(original);
                 needsArtifactory = true;
-                reason = "System property [ fat.test.use.artifactory.substitution ] was set to true, must use Artifactory registry.";
+                reason = "Using a remote docker host, must use Artifactory registry";
                 break;
             }
 
-            // Priority 5: Always use Artifactory if using remote docker host.
-            if (DockerClientFactory.instance().isUsing(EnvironmentAndSystemPropertyClientProviderStrategy.class)) {
-                result = DockerImageName.parse(mirror + '/' + original.asCanonicalNameString())
-                                .withRegistry(ArtifactoryRegistry.instance().getRegistry())
-                                .asCompatibleSubstituteFor(original);
-                needsArtifactory = true;
-                reason = "Using a remote docker host, must use Artifactory registry";
+            // Priority 5: System property artifactory.force.external.repo (NOTE: only honor this property if set to true)
+            if (Boolean.getBoolean(substitutionOverride)) {
+                result = original;
+                needsArtifactory = false;
+                reason = "System property [ fat.test.artifactory.force.external.repo ] was set to true, must use original image name.";
                 break;
             }
```

Comment:
I think this is saying "If we're using a remote docker host, we must use Artifactory" and so we ignore the `force.external` option?

What prompts us to use a remote docker host? Should that decision also respect the `force.external` flag?

Alternatively, is it better for this to remain a separate if you want to be able to enable substituting the repository name without using a remote docker server?

I guess the other issue here is that docker hub is quite strict on their rate limits (unlike maven central) so not using an artifactory proxy is more of an problem.

Reply:
> I think this is saying "If we're using a remote docker host, we must use Artifactory" and so we ignore the force.external option?

Another complex hierarchy of properties and environmental variables.  However, forcing the use of Artifactory when using a remote docker host should be considered before `forceExternal` because we really do not want our external docker hosts to be rate limited.